### PR TITLE
ApplePay Performance

### PR DESCRIPTION
**Pull request by mnicpt**

### CODE REVIEW

File: src/zoid/buttons/component.jsx [hunk 1]
```diff
@@ -67,13 +67,13 @@ export const getButtonsComponent : () => ButtonsComponent = memoize(() => {
...
                 style = {},
-                applePaySupport = isApplePaySupported(),
                 fundingEligibility = getRefinedFundingEligibility(),
                 supportsPopups = userAgentSupportsPopups(),
                 supportedNativeBrowser = isSupportedNativeBrowser(),
                 experiment = getVenmoExperiment(enableVenmoExperiment)
             } = props;
-
+            
+            const applePaySupport = fundingEligibility && fundingEligibility[FUNDING.APPLEPAY]?.eligible ? isApplePaySupported() : false;
```
Comment (gregjopa) [inline src/zoid/buttons/component.jsx:76]: I think you can drop the falsy check. Optional chaining does not blow up when doing `undefined.?`. Ex:

Screen Shot 2021-06-23 at 12 14 30 PM


```suggestion
            const applePaySupport = fundingEligibility[FUNDING.APPLEPAY]?.eligible ? isApplePaySupported() : false;
```

Comment (mnicpt) [reply src/zoid/buttons/component.jsx:76]: When I do that I get lint error.  That's why I added it.

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "description": "PayPal Checkout components, for integrating checkout products.",
     "main": "index.js",
     "scripts": {
-        "dev": "babel-node $(npm bin)/webpack-dev-server --config webpack.config.dev.js --port 9001 --host localhost.paypal.com --open-page demo/dev/index.htm --https --hot=false --inline=false",
+        "dev": "NODE_ENV=test babel-node $(npm bin)/webpack-dev-server --config webpack.config.dev.js --port 9001 --host localhost.paypal.com --open-page demo/dev/index.htm --https --hot=false --inline=false",
```
Comment (bluepnume) [inline package.json:7]: Can we remove this? I don't think we should run dev env in test mode.

File: src/zoid/buttons/component.jsx (hunk 1 above)
Comment (gregjopa) [reply src/zoid/buttons/component.jsx:76]: ahh nevermind then.

File: package.json (hunk 2 above)
Comment (gregjopa) [reply package.json:7]: @mnicpt I took a look at the webpack config and I think we can remove `NODE_ENV=test` from here. And to fix the error, we can add `__PAYPAL_DOMAIN__` and give it a proper value for local development: https://github.com/paypal/paypal-checkout-components/blob/master/webpack.config.dev.js#L30
```
__PAYPAL_DOMAIN__:   'https://www.sandbox.paypal.com',
```

File: src/zoid/buttons/component.jsx (hunk 1 above)
Comment (mnicpt) [reply src/zoid/buttons/component.jsx]: Using now in funding config.

File: test/integration/tests/button/props.js [hunk 3]
```diff
@@ -17,8 +17,7 @@ describe(`paypal button component props`, () => {
         destroyTestContainer();
     });
 
-    it('should render an Apple Pay button if applePaySupport is true', () => {
-        // setup applePaySupport
+    it.only('should render an Apple Pay button if eligible', () => {
```
Comment (gregjopa) [inline test/integration/tests/button/props.js:20]:
```suggestion
    it('should render an Apple Pay button if eligible', () => {
```

File: src/funding/applepay/config.jsx [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
 import { PLATFORM } from '@paypal/sdk-constants/src';
 import { ApplePayLogo, LOGO_COLOR } from '@paypal/sdk-logos/src';
+import { isApplePaySupported } from 'belter';
```
Comment (bluepnume) [inline src/funding/applepay/config.jsx:6]: Gotta use `belter/src` to get tree shaking

**Pull request closed without merging**